modules/server/src/mapping/masking.ts

```diff
@@ -0,0 +1,92 @@
+type Bucket = {
+	doc_count: number;
+	key: string;
+	belowThreshold: boolean;
+};
+
+type Aggregation = {
+	bucket_count: number;
+	buckets: Bucket[];
+};
+
+/**
+ * This returns a total count that is less than or equal to the actual total hits in the query.
+ * It is calculated by adding +1 for values under threshold and bucket.doc_count
+ * for values greater than or equal to
+ *
+ * @param aggregation an aggregation with the most buckets which has data masking applied
+ * @returns hits total value
+ */
+const calculateHitsFromAggregation = ({ aggregation }: { aggregation: Aggregation }) => {
+	return aggregation.buckets.reduce(
+		(totalAcc, bucket) => (bucket.belowThreshold ? totalAcc + 1 : totalAcc + bucket.doc_count),
+		0,
+	);
+};
+
+/**
+ *
+ * 1) Iterate through aggs applying data masking to buckets if applicable
+ * 2) Find the agg with the most bucket count and data masking applied to be used in calculating hits.total
+ *
+ * @param aggregations - aggregations from query
+ * @param thresholdMin - threshold value
+ * @returns aggregations with data masking applied and hits total
+ */
+export const applyAggregationMasking = ({
+	aggregations,
+	thresholdMin,
+}: {
+	aggregations: Record<
+		string,
+		{
+			bucket_count: number;
+			buckets: Array<{
+				doc_count: number;
+				key: string;
+			}>;
+		}
+	>;
+	thresholdMin: number;
+}) => {
+	const THRESHOLD_REPLACEMENT_VALUE = thresholdMin - 1;
```

Comment:
this merits an explanation comment

Reply:
https://github.com/overture-stack/arranger/pull/909/commits/4581e2bd69b8b8f6a4bfe7424d06bd2692ec3af0

Reply:
The code and comment both say that this value is set to one less than the thresholdMin, but why? What is the reason for this?

I guess while I am reading I don't know what a threshold replacement value is used for, and from the threshold rules I remember buckets being set to threshold value or 1. I don't remember the one less than rule.